services/outputhost/ackmanager.go

```diff
@@ -112,75 +119,40 @@ func newAckManager(
 // ackID is a string which is a base64 encoded string
 // First we get the ackID and store the address locally in our data structure
 // for maintaining the ack level
-func (ackMgr *ackManager) getNextAckID(address int64, sequence common.SequenceNumber) (ackID string) {
+func (ackMgr *ackManager) getNextAckID(address storeHostAddress, sequence common.SequenceNumber) (ackID string) {
 	ackMgr.lk.Lock()
+	defer ackMgr.lk.Unlock()
+
 	ackMgr.readLevel++ // This means that the first ID is '1'
+	ackMgr.readLevelSeq++
 
-	expectedReadLevel := ackMgr.levelOffset + ackMgr.readLevel
+	if sequence != ackMgr.readLevelSeq {
 
-	// NOTE: sequence should be zero for timer destinations, since they usually have discontinuous sequence numbers
-	if sequence != 0 && expectedReadLevel != sequence {
-		skippedMessages := sequence - expectedReadLevel
+		ackMgr.logger.WithFields(bark.Fields{
+			`old-readLevelSeq`: ackMgr.readLevelSeq,
+			`new-readLevelSeq`: sequence,
+		}).Error(`adjusting read-level sequence`)
```

Comment:
Let's make this a Warn level log and let's also record a metric here for easy tracking..